app/flatpak-builtins-config.c

```diff
@@ -51,12 +51,12 @@ looks_like_a_language (const char *s)
   int len = strlen (s);
   int i;
 
-  if (len < 2 || len > 3)
+  if (len < 2 || len > 6)
     return FALSE;
 
   for (i = 0; i < len; i++)
     {
-      if (!g_ascii_isalpha (s[i]) || !g_ascii_islower (s[i]))
+      if (!g_ascii_isalpha (s[i]) && !g_strcmp0 (&s[i], "_"))
```

Comment:
This will accept `______` as a valid locale.

Reply:
Oh, you are correct, will think of a better test.

Reply:
This is a documented interface (see flatpak-config.1). Why should we break this? What is the gain?

The main use of the language keys is an optimization - we avoid downloading _all_ the translations, and restrict ourselves to just the ones that are relevant on the local system. For this optimization to work, it is not essential to nail down the locale as precisely as we can - it is sufficient that avoid the bulk of irrelevant locales. If we download en_GB in addition to en_US, that is not such a big deal.

Reply:
The main idea here was to make this setting match a translation information on gnome-software. For example, if I had an app only relevant to english speaking indians, we could match en_IN available through this configuration with the en_IN that comes on gnome-software. As of now, this makes more sense to endless repositories, where we currently store the translations available plainly on the app id (e.g. com.endlessm.biology.en), but we're trying to push this upstream, as a part of the appstream information. See https://lists.freedesktop.org/archives/appstream/2019-August/000309.html